beacon-chain/node/node.go

```diff
@@ -255,6 +255,9 @@ func New(cliCtx *cli.Context, cancel context.CancelFunc, opts ...Option) (*Beaco
 
 	if features.Get().EnableLightClient {
 		beacon.lcStore = lightclient.NewLightClientStore(beacon.db, beacon.fetchP2P(), beacon.StateFeed())
+		if beacon.lcStore == nil {
+			return nil, errors.New("failed to create light client store")
```

Comment:
Since you expect the cache to always be valid, it will be more idiomatic if you return an error directly from `NewLightClientStore`